kingpin/actors/aws/ecs.py

```diff
@@ -366,87 +422,528 @@ def _tasks_done(self, tasks):
+        if service_name is None:
+            service_name = family
+        return service_name
+
+    @gen.coroutine
+    def _describe_service(self, service_name):
+        """Describe a service by name.
+
+        Args:
+            service_name: service name to describe.
+
+        Returns:
+            Service param dict, or None if the service does not exist.
+        """
+        response = yield self.thread(
+            self.ecs_conn.describe_services,
+            cluster=self.option('cluster'),
+            services=[service_name])
+        self._handle_failures(response['failures'], self.FAILURE_MISSING)
+
+        services = response['services']
+
+        # There should never be more than one service for a given name.
+        if len(services) > 1:
+            raise exceptions.UnrecoverableActorFailure(
+                'API returned more than one service for {name}'.format(
+                    name=service_name))
+
+        if services:
+            raise gen.Return(services[0])
+
+    @gen.coroutine
+    @dry('Would create service.')
+    def _create_service(self, service_name, task_definition_name,
+                        client_token):
+        """Create a service.
+
+        Args:
+            service_name: service name to use.
+            task_definition_name: Task Definition string.
+            client_token: uuid for the service creation.
+        """
+        create_parameters = dict(
+            cluster=self.option('cluster'),
+            serviceName=service_name,
+            taskDefinition=task_definition_name,
+            desiredCount=self.option('count'),
+            clientToken=client_token,
+            **self.service_definition)
+
+        self.log.info('Creating service.')
+
+        try:
+            yield self.thread(
+                self.ecs_conn.create_service,
+                **create_parameters)
+        except Boto3Error as e:
+            raise exceptions.UnrecoverableActorFailure(e.message)
+
+    @gen.coroutine
+    @dry('Would update service.')
+    def _update_service(self, service_name, task_definition_name):
+        """Update a service.
+
+        Args:
+            service_name: service name to use.
+            task_definition_name: Task Definition string.
+        """
+        deployment_configuration = self.service_definition[
+            'deploymentConfiguration']
+        update_parameters = dict(
+            cluster=self.option('cluster'),
+            service=service_name,
+            taskDefinition=task_definition_name,
+            desiredCount=self.option('count'),
+            deploymentConfiguration=deployment_configuration)
+
+        self.log.info('Updating service.')
+
+        try:
+            yield self.thread(
+                self.ecs_conn.update_service,
+                **update_parameters)
+        except Boto3Error as e:
+            raise exceptions.UnrecoverableActorFailure(e.message)
+
+    @gen.coroutine
+    @dry('Would ensure the service is registered.')
```

Comment:
This method actually shouldn't be dry -- if it is, the user will never know exactly what changes are going to be made. The underlying API call methods (`self._update_service()`, `self._create_service()`) _do_ have the `@dry` decorator, so they will properly tell the user whats going to happen.